Code/src/main/java/nl/utwente/group10/ui/components/blocks/input/DisplayBlock.java

```diff
@@ -24,102 +25,90 @@
-
+    @FXML protected Pane outputSpace;
+    
     /**
      * Creates a new instance of DisplayBlock.
-     *
      * @param pane
      *            The pane on which this DisplayBlock resides.
      */
     public DisplayBlock(CustomUIPane pane) {
         this(pane, "DisplayBlock");
     }
+    
     protected DisplayBlock(CustomUIPane pane, String fxml) {
         super(pane);
-
-        output = new SimpleStringProperty("New Output");
+        output = new SimpleStringProperty("???");
 
         this.loadFXML(fxml);
 
-        inputAnchor = new InputAnchor(this, pane);
-        anchorSpace.getChildren().add(inputAnchor);
+        inputAnchor = new InputAnchor(this);
+        inputAnchor.layoutXProperty().bind(inputSpace.widthProperty().divide(2));
+        inputSpace.getChildren().add(inputAnchor);
+        
+        //Make sure inputSpace is drawn on top.
+        BorderPane borderPane = (BorderPane) inputSpace.getParent();
+        borderPane.getChildren().remove(inputSpace);
+        borderPane.setTop(inputSpace);
     }
 
     /**
-     * Sets the output flowing into the DisplayBlock and refresh the display.
-     *
-     * @param value
-     *            The value to show.
+     * @return The output this Block is displaying.
      */
-    public void setOutput(final String value) {
-        output.set(value);
+    public String getOutput() {
+        return output.get();
     }
 
     /**
-     * Returns the output value this Block has.
-     * @return outputValue
+     * Sets the output that is displayed.
      */
-    public String getOutput() {
-        return output.get();
+    public void setOutput(final String value) {
+        output.set(value);
     }
 
     /**
-     * Property getter for the output property.
-     * @return outputProperty
+     * @return The StringProperty containing the output.
      */
     public StringProperty outputProperty() {
         return output;
     }
 
     @Override
-    public final Expr asExpr() {
-        return inputAnchor.asExpr();
-    }
-
-    /** Invalidates the outputted value and triggers re-evaluation of the value. */
-    public void invalidateConnectionState() {
+    public void invalidateVisualState() {
+        super.invalidateVisualState();
         try {
             Optional<GhciSession> ghci = getPane().getGhciSession();
 
             if (ghci.isPresent()) {
-                setOutput(ghci.get().pull(inputAnchor.asExpr()));
+                setOutput(ghci.get().pull(inputAnchor.getExpr()));
             }
         } catch (GhciException e) {
             setOutput("???");
         }
     }
-
+    
     @Override
-    public Type getInputType(InputAnchor anchor) {
-        if(anchor.getPrimaryOppositeAnchor().isPresent()) {
-            return anchor.getPrimaryOppositeAnchor().get().getType();
-        } else {
-            return getInputSignature(anchor);
-        }
+    public List<InputAnchor> getAllInputs() {
+        return ImmutableList.of(inputAnchor);
```

Comment:
They do not always as far as I can determine, for example in FunctionBlock. I am not going to fix this as I am not aware of the complications.